src/Kiota.Builder/Writers/Php/CodeMethodWriter.cs

```diff
@@ -346,10 +346,11 @@ private void WriteRequestGeneratorBody(CodeMethod codeElement, RequestParams req
                 var suffix = requestParams.requestBody.Type.IsCollection ? "Collection" : string.Empty;
                 if (requestParams.requestBody.Type.Name.Equals(conventions.StreamTypeName, StringComparison.OrdinalIgnoreCase))
                     writer.WriteLine($"{RequestInfoVarName}->setStreamContent({conventions.GetParameterName(requestParams.requestBody)});");
-                if (requestParams.requestBody.Type is CodeType bodyType && bodyType.TypeDefinition is CodeClass) {
+                else if (requestParams.requestBody.Type is CodeType bodyType && bodyType.TypeDefinition is CodeClass) {
                     writer.WriteLine($"{RequestInfoVarName}->setContentFromParsable{suffix}($this->{requestAdapterProperty.Name.ToFirstCharacterLowerCase()}, \"{codeElement.RequestBodyContentType}\", {conventions.GetParameterName(requestParams.requestBody)});");
+                } else {
+                    writer.WriteLine($"{RequestInfoVarName}->setContentFromScalar{suffix}($this->{requestAdapterProperty.Name.ToFirstCharacterLowerCase()}, \"{codeElement.RequestBodyContentType}\", {conventions.GetParameterName(requestParams.requestBody)});");
```

Comment:
As it is right now, if it is not a stream or categorized as parseable, it will always assume it is a scalar value. What happens if the content is plain text?  Would it make sense to check whether it is a scalar value in an `else if` and then throw something like a NotSupportException if is not a scalar value?